samples/utils.py

```diff
@@ -0,0 +1,71 @@
+import time
+from socket import timeout as Timeout
+# [START get_service]
+from oauth2client.client import GoogleCredentials
+from googleapiclient.discovery import build
+
+
+def get_service():
+    credentials = GoogleCredentials.get_application_default()
+    return build('bigquery', 'v2', credentials=credentials)
+# [END get_service]
+
+
+# [START poll_job]
+def poll_job(service, projectId, jobId, timeout=1, max_timeout=33):
+    job_get = service.jobs().get(
+            projectId=projectId,
+            jobId=jobId)
+    job_resource = job_get.execute()
+
+    while not job_resource['status']['state'] == 'DONE':
+        if timeout > max_timeout:
+            raise StandardError(
+                    'Timed out waiting for job {} to complete, '
+                    'after {} seconds'
+                    .format(jobId, timeout-1))
+
+        print('Job is {}, waiting {} seconds...'
+              .format(job_resource['status']['state'], timeout))
+
+        time.sleep(timeout)
+        timeout *= 2
+        job_resource = job_get.execute()
```

Comment:
BTW there is no mention of a num_retries kwarg in the docs so if it does exist, it needs to be documented